source/code/modules/iceshard_pipelines/private/asset_font.cxx

```diff
@@ -0,0 +1,234 @@
+﻿#include "asset_font.hxx"
+
+#include <ice/uri.hxx>
+#include <ice/resource.hxx>
+#include <ice/resource_meta.hxx>
+#include <ice/resource_tracker.hxx>
+#include <ice/task_sync_wait.hxx>
+#include <ice/font.hxx>
+
+#include <msdf-atlas-gen/msdf-atlas-gen.h>
+
+
+namespace ice
+{
+
+    auto create_engine_object(
+        ice::Allocator& alloc,
+        std::vector<msdf_atlas::GlyphGeometry> const& glyphs,
+        msdfgen::BitmapConstRef<msdfgen::byte, 4> const& bitmap
+    ) noexcept -> ice::Memory
+    {
+        using ice::Font;
+        using ice::FontAtlas;
+        using ice::GlyphRange;
+        using ice::Glyph;
+
+        static_assert(alignof(Font) == 8);
+        static_assert(alignof(FontAtlas) == 4);
+        static_assert(alignof(GlyphRange) == 4);
+
+        ice::u32 size = sizeof(Font) + sizeof(FontAtlas) + sizeof(GlyphRange) + 8 /* for alignment */;
+        size += sizeof(Glyph) * glyphs.size();
+        size += bitmap.width * bitmap.height * 4;
+
+        void* data = alloc.allocate(size, 8);
+
+        Font* gfx_font = reinterpret_cast<Font*>(data);
+        FontAtlas* gfx_atlas = reinterpret_cast<FontAtlas*>(
+            ice::memory::ptr_align_forward(gfx_font + 1, alignof(FontAtlas))
+        ); // TODO: Multi-atlas support
```

Comment:
Create a task for font-atlas support and another for multiple atlasses in one font asset.

Reply:
Created #132